docs/plugins/code_deploy_setup.md

```diff
@@ -0,0 +1,60 @@
+# Moonshot CodeDeploy Setup plugin
+Moonshot plugin to ensure CodeDeploy has all the necessary S3 buckets created
+ to support deploying to multiple configured regions.
+
+## Moonshot configuration
+Update your Moonfile.rb with the following configuration
+
+```ruby
+Moonshot.config do |c|
+  # ...
+  c.app_name = 'example-project'
+
+  bucket_name = "#{c.app_name}-builds"
+
+  # bucket_name: Base name of the S3 bucket to create, will have region appended to it
+  # prefix: Optional. If using the same bucket for multiple apps then recommend adding a prefix
+  # regions: list of all supported regions you would like to use. Default: ENV['AWS_REGION']
+  code_deploy_setup_plugin = Moonshot::Plugins::CodeDeploySetup.new(
+    bucket_name,
+    prefix: c.app_name,
+    regions: [ 'us-east-1', 'us-west-2' ]
```

Comment:
Thought was that the given app would define the specific regions it ran in, CDB has an AWS helper that has the list of their specific supported regions that get passed into this plugin